datafusion/core/tests/sqllogictests/src/engines/conversion.rs

```diff
@@ -74,9 +74,9 @@ pub fn f64_to_str(value: f64) -> String {
     }
 }
 
-pub fn i128_to_str(value: i128, scale: u32) -> String {
+pub fn i128_to_str(value: i128, precision: &u8, scale: &i8) -> String {
     big_decimal_to_str(
-        BigDecimal::from_str(&Decimal::from_i128_with_scale(value, scale).to_string())
```

Comment:
`rust-decimal` doesn't allow precision > 28. We can simply use `Decimal128Type` from arrow-rs to get string of decimal.